### ISSUE TITLE
Multiple TR_ALLOC definition warning

### ISSUE THREAD

**Issue by 0xdaryl** [eclipse/omr#1699]
Building Tril compiler tests (e.g., on x86-64) yields the following warning:

```
In file included from /home/travis/build/eclipse/omr/compiler/il/OMRNode.hpp:40:0,
                 from /home/travis/build/eclipse/omr/compiler/il/Node.hpp:25,
                 from /home/travis/build/eclipse/omr/fvtest/compilertriltest/SimplifierFoldAndTest.cpp:24:
/home/travis/build/eclipse/omr/compiler/env/TRMemory.hpp:707:0: warning: "TR_ALLOC" redefined [enabled by default]
 #define TR_ALLOC(a) \
 ^
In file included from /home/travis/build/eclipse/omr/compiler/ilgen/IlBuilder.hpp:33:0,
                 from /home/travis/build/eclipse/omr/compiler/ilgen/TypeDictionary.hpp:38,
                 from /home/travis/build/eclipse/omr/fvtest/tril/tril/ilgen.hpp:25,
                 from /home/travis/build/eclipse/omr/fvtest/tril/tril/method_info.hpp:26,
                 from /home/travis/build/eclipse/omr/fvtest/tril/tril/method_compiler.hpp:25,
                 from /home/travis/build/eclipse/omr/fvtest/tril/tril/jitbuilder_compiler.hpp:25,
                 from /home/travis/build/eclipse/omr/fvtest/compilertriltest/SimplifierFoldAndTest.cpp:23:
/home/travis/build/eclipse/omr/compiler/ilgen/IlInjector.hpp:56:0: note: this is the location of the previous definition
 #define TR_ALLOC(x)
 ^
```

Although `TRMemory.hpp` itself is guarded for multiple includes, the macros defined within it are not.  I'm not convinced that protecting those macros from multiple inclusion is the right answer yet.  This will need some investigation.

**Comment by Leonardo2718** [comment]
Actually, I believe this happens when building JitBuilder in general, not just Tril.

**Comment by mstoodle** [comment]
Yeah, it's because these files are also used independently of the JIT (i.e. in jitbuilder/release/include). Rather than slurping in the entire TR_Memory set of headers, I just provided this escape hatch to define the macro (since it isn't neeed in the other scenario). I could probably have wrapped in a #ifndef block :) .

**Comment by 0xdaryl** [comment]
This is fixed by #3596.
